dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -1,5 +1,322 @@
-{% materialization snapshot, adapter='athena' -%}
-  {{ exceptions.raise_not_implemented(
-    'snapshot materialization not implemented for '+adapter.type())
-  }}
+{%
+  Hash function to generate dbt_scd_id. dbt by default uses md5(coalesce(cast(field as varchar)))
+   but it does not work with athena. It throws an error Unexpected parameters (varchar) for function
+   md5. Expected: md5(varbinary)
+%}
+
+{% macro athena__snapshot_hash_arguments(args) -%}
+    to_hex(md5(to_utf8({%- for arg in args -%}
```

Comment:
We could also do just `md5(to_utf8(coalesce(cast(<field> as varchar))))` but I include `to_hex` for easier visual verification. If its better to leave it with `md5()` then I can change that.

Reply:
This looks good, actually `to_hex(md5(to_utf8(field)))` it's a nice macro. I won't include coalesce but I will leave the liberty to the user to cover edge cases in their models.

Reply:
I included coalesce because I wanted to retain whatever logic dbt used in their hash macro so that it remains consistent